queries/userinfo.py

```diff
@@ -0,0 +1,400 @@
+        with BIN_Db.get_db() as bdb:
+            fn = components[0].title()
+            gender = bdb.lookup_name_gender(fn) or "hk"
+            answ = _MY_NAME_IS_RESPONSES[gender].format(fn)
+
+        # Save this info about user to query data table
+        if q.client_id:
+            qdata = dict(full=name.title(), first=fn, gender=gender)
+            q.set_client_data("name", qdata)
+
+        # Generate answer
+        voice = answ.replace(",", "")
+        q.set_answer(dict(answer=answ), answ, voice)
+        q.query_is_command()
+        return True
+
+    return False
+
+
+def _addr2str(addr: Dict[str, str], case: str = "nf") -> str:
+    """ Format address canonically given dict w. address info. """
+    assert case in ["nf", "þgf"]
+    prep = iceprep_for_placename(addr["placename"])
+    astr = "{0} {1} {2} {3}".format(
+        addr["street"], addr["number"], prep, addr["placename"]
+    )
+    if case == "þgf":
+        try:
+            n = NounPhrase(astr)
+            if n:
+                astr = n.dative or astr
+        except Exception:
+            pass
+    return numbers_to_neutral(astr)
+
+
+_WHATS_MY_ADDR = frozenset(
+    (
+        "hvar á ég heima",
+        "hvar á ég eiginlega heima",
+        "veistu hvar ég á heima",
+        "veist þú hvar ég á heima",
+        "hvar bý ég",
+        "hvar bý ég eiginlega",
+        "veistu hvar ég bý",
+        "veist þú hvar ég bý",
+        "hvað er heimilisfang mitt",
+        "hvað er heimilisfangið mitt",
+        "hvert er heimilisfang mitt",
+        "hvert er heimilisfangið mitt",
+    )
+)
+
+
+_DUNNO_ADDRESS = "Ég veit ekki hvar þú átt heima, en þú getur sagt mér það."
+
+
+def _whatsmyaddr_handler(q: Query, ql: str) -> bool:
+    """ Handle queries of the form "Hvar á ég heima?" """
+    if ql in _WHATS_MY_ADDR:
+        answ = None
+        ad = q.client_data("address")
+        if not ad:
+            answ = _DUNNO_ADDRESS
+        else:
+            prep = iceprep_for_street(ad["street"])
+            answ = "Þú átt heima {0} {1}".format(prep, _addr2str(ad, case="þgf"))
+        q.set_answer(*gen_answer(answ))
+        return True
+
+
+_MY_ADDRESS_REGEXES = (
+    r"ég á heima á (.+)$",
+    r"ég á heima í (.+)$",
+    r"heimilisfang mitt er á (.+)$",
+    r"heimilisfang mitt er í (.+)$",
+    r"heimilisfang mitt er (.+)$",
+)
+
+_ADDR_LOOKUP_FAIL = "Ekki tókst að fletta upp þessu heimilisfangi."
+
+
+def _myaddris_handler(q: Query, ql: str) -> bool:
+    """ Handle queries of the form "Ég á heima á [heimilisfang]".
+        Store this info as query data. """
+    for rx in _MY_ADDRESS_REGEXES:
+        m = re.search(rx, ql)
+        if m:
+            break
+    if not m:
+        return False
+
+    addr_str = m.group(1).strip()
+    if not addr_str:
+        return False
+
+    # Try to parse address, e.g. "Öldugötu 4 [í Reykjavík]"
+    m = re.search(r"^(\w+)\s(\d+)\s?([í|á]\s)?(\w+)?$", addr_str.strip())
```

Comment:
This does not support "Vatnsstígur 4b". Actually, the parser could be used for this - and, come to think of it, a dedicated Greynir class ("Address"), similar to NounPhrase, would be useful here and in other use cases.